### PR TITLE
Evaluate LM on Linzen task

### PR DESCRIPTION
**Pull request by LK**
I get accuracy of ~53% with this, while Linzen reports >93% accuracy on their LM. Quite possible that there's a bug in here somewhere that I missed, but I haven't been able to find anything — would be good to get another pair of eyes on this to see if I'm missing something.

### CODE REVIEW

File: stack_rnn_LM.py [hunk 1]
```diff
@@ -91,6 +92,18 @@ def forward(self, sentence, label=None):
 
         stacked_h = torch.stack(h_all_words, dim=1)
         logits = self._classifier(stacked_h)
+
+        # TODO: there must be a better way to do this?
+        final_logits_mask = np.zeros((mask.shape[0],))
```
Comment (LK) [inline stack_rnn_LM.py:98]: This block is to deal with the mask; replacing this entire chunk with `final_logits = logits[:,-1,:]` (which will incorrectly include padding sometimes) barely affects performance (53 -> 55)

File: linzen_lm_evaluate.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        logits = res["final_logits"]
+        grammatical_labels = batch["label"]
+        grammatical_words = [vocab.get_token_from_index(x.item()) for x in grammatical_labels]
+        ungrammatical_words = [inflect[x] for x in grammatical_words]
+        ungrammatical_labels = [vocab.get_token_index(x) for x in ungrammatical_words]
+    
+        for i in range(16):
+            # print([vocab.get_token_from_index(x.item()) for x in batch["sentence"]["tokens"][i]])
+            # print(grammatical_words[i], ungrammatical_words[i])
+            # print(logits[i][grammatical_labels[i]].item(), logits[i][ungrammatical_labels[i]].item())
```
Comment (LK) [inline linzen_lm_evaluate.py]: uncomment these to see input, grammatical/ungrammatical next prediction, and predicted probabilities for each sentence

File: linzen_lm_evaluate.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    test_dataset = reader.read("data/agr_50_mostcommon_10K.tsv.10")
+    vocab = Vocabulary.from_files("saved_models/vocabulary-linzen")
+
+    model = StackRNNLanguageModel(vocab, rnn_dim=100, stack_dim=16)
+    model.load_state_dict(torch.load("saved_models/stack-linzen-swap.th"))
+
+    iterator = BucketIterator(batch_size=16, sorting_keys=[("sentence", "num_tokens")])
+    iterator.index_with(vocab)
+
+    inflect, _ = utils.gen_inflect_from_vocab()
```
Comment (LK) [inline linzen_lm_evaluate.py:31]: this comes from linzen's repository; looks correct from a quick skim through its contents
